com.unity.shadergraph/Editor/Generation/Processors/ShaderSpliceUtil.cs

```diff
@@ -321,6 +321,16 @@ private bool ProcessSpliceCommand(Token spliceCommand, int lineEnd, ref int cur)
                 return true;
             }
 
+            private bool ContainsCommand(string source, int begin, int end)
+            {
+                for (int i = begin; i < end && i < source.Length; ++i)
+                {
+                    if (source[i] == '$')
+                        return true;
+                }
+                return false;
```

Comment:
You may try to use a new C# syntax:
```csharp
return source[begin..end].Contains("$");
```

Reply:
It's technically valid (I tried locally, I wasn't aware of this new syntax, it's nice ❤️), however :
- `source[begin..end]` seems to return a string allocated on the heap. Is it equivalent with `Substring` or is it an Enumerable<char>/Span<char>?. This ContainsCommand is called for every line of the shader template.
- This file is free of any Linq, it even uses custom code to speed up SG generation (see https://github.com/Unity-Technologies/Graphics/pull/3305 from @elizabeth-legros), I guess the goal of this implementation is to be the most explicit as possible.
I would like to stay consistent with the chosen approach in this API.

Reply:
I can understand, no problem :)